packages/nouns-subgraph/src/custom-types/ParsedProposalV3.ts

```diff
@@ -0,0 +1,102 @@
+import { BigInt, Bytes } from '@graphprotocol/graph-ts';
+import {
+  ProposalCreatedWithRequirements,
+  ProposalCreatedWithRequirements1,
+} from '../types/NounsDAO/NounsDAO';
+import { BIGINT_ZERO, STATUS_ACTIVE, STATUS_PENDING } from '../utils/constants';
+
+export class ParsedProposalV3 {
+  id: string = '';
+  proposer: string = '';
+  txHash: string = '';
+  targets: Bytes[] = [];
+  values: BigInt[] = [];
+  signatures: string[] = [];
+  calldatas: Bytes[] = [];
+  createdTimestamp: BigInt = BIGINT_ZERO;
+  createdBlock: BigInt = BIGINT_ZERO;
+  createdTransactionHash: Bytes = Bytes.fromI32(0);
+  startBlock: BigInt = BIGINT_ZERO;
+  endBlock: BigInt = BIGINT_ZERO;
+  proposalThreshold: BigInt = BIGINT_ZERO;
+  quorumVotes: BigInt = BIGINT_ZERO;
+  description: string = '';
+  title: string = '';
+  status: string = '';
+  signers: string[] = [];
+
+  static fromV1Event(event: ProposalCreatedWithRequirements1): ParsedProposalV3 {
+    const proposal = new ParsedProposalV3();
+
+    proposal.id = event.params.id.toString();
```

Comment:
@eladmallel consider extracting all the parts identical to the function below

Reply:
@davidbrai the main issue is that these two event types are auto generated without a shared super type with all the common fields, that's why I wrote out these two separate functions. also note that the way event params are handled is by using an array inside where each field is a certain index, and they don't overlap in the autogenerated code, making it even harder to refactor and reuse, so I just opted for code duplication.

wdyt?

Reply:
@eladmallel yeah I see the issue. I guess assemblyscript is pretty rigid on the types